core/config/toml/types.go

```diff
@@ -1455,6 +1455,7 @@ type Tracing struct {
 	CollectorTarget *string
 	NodeID          *string
 	SamplingRatio   *float64
+	TLSCertPath     *string
```

Comment:
should this be omitted if empty? iiuc a user can omit this if/when insecure mode is desired

thinking more about it, i have a sense that it would be better to force a user to explicitly disable the feature rather than implicitly disabling it by omission.

how do other systems deal with this? i know postgres you have to set it to insecure mode...

would a struct work here like ~

TLSConfig struct {
mode string 
certpath string # ignored if mode is insecure
}?

Reply:
Nice idea - I like explicitly configuring an insecure mode since it makes it very obvious to the user what is going on.

Reply:
> should this be omitted if empty? iiuc a user can omit this if/when insecure mode is desired
Yes, this is the behavior I want. 

I'm having difficulty getting gotoml to play nicely though. It seems to marshal it to an empty string even when not set by the user, which is undesirable. To Jordan's point, setting `omitempty` has no effect on pointers.